engine/src/main/java/org/hibernate/validator/internal/engine/ValidatorContextProperties.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.internal.engine;
+
+import java.time.Duration;
+
+import javax.validation.ClockProvider;
+import javax.validation.MessageInterpolator;
+import javax.validation.TraversableResolver;
+import javax.validation.Validator;
+
+import org.hibernate.validator.internal.engine.resolver.TraversableResolvers;
+import org.hibernate.validator.internal.util.ExecutableParameterNameProvider;
+import org.hibernate.validator.spi.scripting.ScriptEvaluatorFactory;
+
+/**
+ * Context object keeping storing all possible validator configurable properties.
+ * There should be one per {@link Validator} instance.
+ *
+ * @author Marko Bekhta
+ */
+public class ValidatorContextProperties {
```

Comment:
I don't like the name but I can see how it is difficult to find a good name given we already have a `ValidatorContext` in `javax.validation`. I can't think of anything better than `ValidatorContext` though.

That being said, if we take into account my remark about limiting the class to `ValidatorImpl` and `ValidationContext`, I think that might be acceptable to name it `ValidatorContext` as the class will be very localized.

@gunnarmorling thoughts?

Reply:
Just needed to use something different than `ValidatorContext` :). How about `InstanceScopeValidatorContext` ?

Reply:
Yuk :).

Maybe `ValidatorScopedContext`? And maybe we make it a package protected inner class of `ValidationContext`and reference it as `ValidationContext.ValidatorScopedContext`?

That's the best I can think of right now.